app/models/spree/gateway/stripe_connect.rb

```diff
@@ -0,0 +1,147 @@
+module Spree
+  class Gateway::StripeConnect < Gateway
+    preference :enterprise_id, :integer
+
+    attr_accessible :preferred_enterprise_id
+
+    CARD_TYPE_MAPPING = {
+      'American Express' => 'american_express',
+      'Diners Club' => 'diners_club',
+      'Visa' => 'visa'
+    }
+
+    def method_type
+      'stripe'
+    end
+
+    def provider_class
+      ActiveMerchant::Billing::StripeGateway
+    end
+
+    def payment_profiles_supported?
+      true
+    end
+
+    def stripe_account_id
+      StripeAccount.find_by_enterprise_id(preferred_enterprise_id).andand.stripe_user_id
+    end
+
+    def purchase(money, creditcard, gateway_options)
+      provider.purchase(*options_for_purchase_or_auth(money, creditcard, gateway_options))
+    end
+
+    def authorize(money, creditcard, gateway_options)
+      provider.authorize(*options_for_purchase_or_auth(money, creditcard, gateway_options))
+    end
+
+    def capture(money, response_code, gateway_options)
+      provider.capture(money, response_code, gateway_options)
+    end
+
+    def credit(money, creditcard, response_code, gateway_options)
+      provider.refund(money, response_code, {})
+    end
+
+    def void(response_code, creditcard, gateway_options)
+      provider.void(response_code, {})
+    end
+
+    def cancel(response_code)
+      provider.void(response_code, {})
+    end
+
+    def create_profile(payment)
+      return unless payment.source.gateway_customer_profile_id.nil?
+      options = {
+        email: payment.order.email,
+        login: Stripe.api_key,
+      }.merge! address_for(payment)
+
+      source = update_source!(payment.source)
+      if source.number.blank? && source.gateway_payment_profile_id.present?
+        creditcard = source.gateway_payment_profile_id
+      else
+        creditcard = source
+      end
+
+      response = provider.store(creditcard, options)
+      if response.success?
+        payment.source.update_attributes!({
+          cc_type: payment.source.cc_type, # side-effect of update_source!
+          gateway_customer_profile_id: response.params['id'],
+          gateway_payment_profile_id: response.params['default_source'] || response.params['default_card']
+        })
+
+      else
+        payment.send(:gateway_error, response.message)
+      end
+    end
```

Comment:
This method seems too long to me, I see the opportunity to split it.

Reply:
TODO

Reply:
Resolved